kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/GcpClientBuilderIT.java

```diff
@@ -0,0 +1,67 @@
+package com.wepay.kafka.connect.bigquery.integration;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.DatasetId;
+import com.google.cloud.storage.Storage;
+import com.wepay.kafka.connect.bigquery.GcpClientBuilder;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig;
+import org.apache.kafka.test.IntegrationTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+
+@Category(IntegrationTest.class)
+public class GcpClientBuilderIT extends BaseConnectorIT {
+
+    private BigQuerySinkConfig connectorProps(GcpClientBuilder.KeySource keySource) throws IOException {
+        Map<String, String> properties = baseConnectorProps(1);
+        properties.put(BigQuerySinkConfig.KEY_SOURCE_CONFIG, keySource.name());
+
+        if(keySource == GcpClientBuilder.KeySource.APPLICATION_DEFAULT) {
+            properties.put(BigQuerySinkConfig.KEYFILE_CONFIG, null);
+        }
+        else if (keySource == GcpClientBuilder.KeySource.JSON){
+            // actually keySource is the path to the credentials file, so we convert it to the json string

```

Comment:
Nit: should this be `keyFile` instead of `keySource`? Disregard this suggestion if not.

```suggestion
            // actually keyFile is the path to the credentials file, so we convert it to the json string

```